externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/KyuubiSparkUtil.scala

```diff
@@ -93,4 +96,13 @@ object KyuubiSparkUtil extends Logging {
     val runtimeVersion = SemanticVersion(runtimeSparkVer)
     (runtimeVersion.majorVersion, runtimeVersion.minorVersion)
   }
+
+  def parsePlan(spark: SparkSession, plan: LogicalPlan): String = {
+    import org.apache.kyuubi.plugin.lineage.helper.LineageParser
+    val lineageParser =
+      new LineageParser { def sparkSession = spark }
```

Comment:
can we use reflect here ? then we can remove maven dependency

Reply:
Using reflection is similar to setting the dependency of spark-lineage-plugin to `optional` and requires that this dependency logic be put into a separate file to pass all the unit tests.